collector/src/bin/rustc-perf-collector/sysroot.rs

```diff
@@ -82,47 +83,41 @@ impl fmt::Display for ModuleVariant {
             ModuleVariant::Cargo => write!(f, "cargo"),
             ModuleVariant::Rustc => write!(f, "rustc"),
             ModuleVariant::Std => write!(f, "rust-std"),
+            ModuleVariant::RustSrc => write!(f, "rust-src"),
         }
     }
 }
 
 impl ModuleVariant {
     fn url(&self, sysroot: &SysrootDownload, triple: &str) -> String {
-        MODULE_URL
-            .replace("@MODULE@", &self.to_string())
-            .replace("@SHA@", &sysroot.rust_sha)
-            .replace("@TRIPLE@", triple)
+        let suffix = if *self == ModuleVariant::RustSrc {
+            String::new()
+        } else {
+            format!("-{}", triple)
+        };
+        format!(
+            "{base}/{sha}/{module}-nightly{suffix}.tar.xz",
+            base = BASE_URL,
+            module = self,
+            sha = sysroot.rust_sha,
+            suffix = suffix,
+        )
     }
 }
 
 impl SysrootDownload {
     fn into_sysroot(self) -> anyhow::Result<Sysroot> {
+        let sysroot_dir = self.directory.join(&self.rust_sha);
+        let sysroot_dir = sysroot_dir.canonicalize().with_context(|| {
+            format!(
+                "failed to canonicalize sysroot path for {}: {:?}",
+                self.rust_sha, sysroot_dir
+            )
+        })?;
         Ok(Sysroot {
-            rustc: self
-                .directory
-                .join(&self.rust_sha)
-                .join("rustc/bin/rustc")
-                .canonicalize()
-                .with_context(|| {
-                    format!("failed to canonicalize rustc path for {}", self.rust_sha)
-                })?,
-            rustdoc: self
-                .directory
-                .join(&self.rust_sha)
-                .join("rustc/bin/rustdoc")
-                .canonicalize()
-                .with_context(|| {
-                    format!("failed to canonicalize rustdoc path for {}", self.rust_sha)
-                })?,
-            cargo: {
-                let path = self.directory.join(&self.rust_sha).join("cargo/bin/cargo");
-                path.canonicalize().with_context(|| {
-                    format!(
-                        "failed to canonicalize cargo path for {}: {:?}",
-                        self.rust_sha, path
-                    )
-                })?
-            },
+            rustc: sysroot_dir.join("bin/rustc"),
+            rustdoc: sysroot_dir.join("bin/rustdoc"),
+            cargo: sysroot_dir.join("bin/cargo"),
```

Comment:
So this is actually I believe a behavior change because previously we'd be able to detect that rustc/rustdoc/cargo don't exist and now we won't -- could we add some `assert!(path.exists(), "rustc exists");` calls perhaps? I agree this is clearer.

Reply:
Ahhhh that's what that was for, I didn't know canonicalization errored for non-existant paths.